std/typecons.d

```diff
@@ -3109,6 +3109,71 @@ struct Nullable(T)
     {
         return isNull ? fallback : _value.payload;
     }
+
+    // Range interface
+    /**
+     * Returns whether or not any values remain in this range.
+     */
+    alias empty = isNull;
+    /**
+     * Advances this range.
+     */
+    alias popFront = nullify;
+    /// ditto
+    alias popBack = nullify;
+    /**
+     * Returns the range's current value.
+     */
+    @property ref inout(T) front() inout @safe pure nothrow
+    {
+        return get();
+    }
+    /// ditto
+    alias back = front;
+    ///
+    @property inout(typeof(this)) save()() inout
+    {
+        return this;
+    }
+    ///
+    inout(typeof(this)) opSlice()() inout
```

Comment:
This form of `opSlice` is considered outdated and should not be used in new code:

> For backward compatibility, `a[]` and `a[i..j]` can also be overloaded by implementing `opSlice()` with no arguments and `opSlice(i, j)` with two arguments, respectively. This only applies for one-dimensional slicing, and dates from when D did not have full support for multidimensional arrays. This usage of `opSlice` is discouraged.

Please use `opIndex()` instead, as described in the language spec's section on [slice operator overloading.][1]

[1]: https://dlang.org/spec/operatoroverloading.html#slice

Reply:
Oh wow, I completely missed the memo on this change.